src/clj/fluree/db/query/api.cljc

```diff
@@ -64,24 +64,25 @@
 (defn query-fql
   "Execute a query against a database source. Returns core async channel
   containing result or exception."
-  [ds query]
-  (go-try
-    (let [{query :subject, did :did} (or (<? (cred/verify query))
-                                         {:subject query})
-          {:keys [t opts] :as query*} (update query :opts sanitize-query-options did)
+  ([ds query] (query-fql ds query nil))
+  ([ds query {:keys [did issuer] :as _opts}]
+   (go-try
+    ;; TODO - verify if both 'did' and 'issuer' opts are still needed upstream
+    (let [{:keys [t opts] :as query*} (update query :opts sanitize-query-options (or did issuer))
 
           ;; TODO: extracting query context here for policy only to do it later
           ;; while parsing the query. We need to consolidate both policy and
           ;; query parsing while cleaning up the query api call stack.
           q-ctx    (ctx-util/extract query*)
+          ;; TODO - remove restrict-db from here, restriction should happen upstream if needed
```

Comment:
👍🏾

Reply:
Also, once we make the change to restricting the db upstream, we should remove the `opts` argument here and accept `did` and `issuer` directly. I think open ended options maps should be reserved for user facing api functions, and the options maps should be parsed, validated, and destructured immediately at the api surface instead of passed directly to internal functions, and the signatures of non-user facing functions should be explicit.